tests/json-fixtures/test_transactions.py

```diff
@@ -40,68 +45,90 @@
+    elif fork_name == ForkName.Metropolis:
+        pytest.skip("Metropolis Transaction class has not been implemented")
+    else:
+        raise ValueError("Unknown Fork Name: {0}".format(fork_name))
 
-    if 'sender' in fixture:
-        transaction = rlp.decode(fixture['rlp'], sedes=TransactionClass)
-        expected = normalize_signed_transaction(fixture['transaction'])
-
-        assert transaction.nonce == expected['nonce']
-        assert transaction.gas_price == expected['gasPrice']
-        assert transaction.gas == expected['gasLimit']
-        assert transaction.to == expected['to']
-        assert transaction.value == expected['value']
-        assert transaction.data == expected['data']
-        assert transaction.v == expected['v']
-        assert transaction.r == expected['r']
-        assert transaction.s == expected['s']
 
-        sender = to_canonical_address(fixture['sender'])
+def test_transaction_fixtures(fixture, fixture_transaction_class):
 
-        try:
-            assert transaction.sender == sender
-        except BadSignature:
-            assert not (27 <= transaction.v <= 34)
-    else:
-        # check RLP correctness
-        try:
-            transaction = rlp.decode(fixture['rlp'], sedes=TransactionClass)
-        # fixture normalization changes the fixture key from rlp to rlpHex
-        except KeyError:
-            assert fixture['rlpHex']
-            return
-        # rlp is a list of bytes when it shouldn't be
-        except TypeError as err:
-            assert err.args == ("'bytes' object cannot be interpreted as an integer",)
-            return
-        # rlp is invalid or not in the correct form
-        except (rlp.exceptions.ObjectDeserializationError, rlp.exceptions.DecodingError):
-            return
+    TransactionClass = fixture_transaction_class
 
+    try:
+        txn = rlp.decode(fixture['rlp'], sedes=TransactionClass)
+    except (rlp.DeserializationError, rlp.exceptions.DecodingError):
+        assert 'hash' not in fixture, "Transaction was supposed to be valid"
+    except TypeError as err:
+        assert err.args == ("'bytes' object cannot be interpreted as an integer",)
+        assert 'hash' not in fixture, "Transaction was supposed to be valid"
+    # fixture normalization changes the fixture key from rlp to rlpHex
+    except KeyError:
+        assert fixture['rlpHex']
+        assert 'hash' not in fixture, "Transaction was supposed to be valid"
+    else:
         # check parameter correctness
         try:
-            transaction.validate()
+            txn.validate()
         except ValidationError:
             return
+
+    if 'sender' in fixture:
+        assert 'hash' in fixture, "Transaction was supposed to be invalid"
+        sender = to_canonical_address(fixture['sender'])
+
+        try:
+            assert txn.get_sender() == sender
+        except BadSignature:
+            assert not (27 <= txn.v <= 34)
```

Comment:
Can you explain this assertion.  I don't exactly understand what it's validating or why it's here.

Reply:
I can not :) I just kept this one as [we currently have it](https://github.com/ethereum/py-evm/blob/5c273fff1ed8f4815b7232a442061673c00f1c60/tests/json-fixtures/test_transactions.py#L86). I can take it out.

Reply:
Git blame says @DavidKnott [would be the one who could answer it](https://github.com/ethereum/py-evm/commit/f868a6e47256eb334fdb1ed620604ee9390e1af0) ;)

Reply:
lol, maybe see if removing it causes failures?  This looks like it's an [EIP155](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md) related thing, but I don't think that those should cause bad signature failures but I'm not sure about that.  Maybe check what go-ethereum is doing?